internal/services/automation/automation_runtime_environment_resource.go

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package automation
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2024-10-23/runtimeenvironment"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type AutomationRuntimeEnvironmentResourceModel struct {
+	Name                   string            `tfschema:"name"`
+	ResourceGroupName      string            `tfschema:"resource_group_name"`
+	AutomationAccountName  string            `tfschema:"automation_account_name"`
+	RuntimeLanguage        string            `tfschema:"runtime_language"`
+	RuntimeVersion         string            `tfschema:"runtime_version"`
+	RuntimeDefaultPackages map[string]string `tfschema:"runtime_default_packages"`
+	Location               string            `tfschema:"location"`
+	Description            string            `tfschema:"description"`
+	Tags                   map[string]string `tfschema:"tags"`
+}
+
+type AutomationRuntimeEnvironmentResource struct{}
+
+var (
+	_ sdk.ResourceWithUpdate        = AutomationRuntimeEnvironmentResource{}
+	_ sdk.ResourceWithCustomizeDiff = AutomationRuntimeEnvironmentResource{}
+)
+
+func (m AutomationRuntimeEnvironmentResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			rd := metadata.ResourceDiff
+
+			if rd.HasChange("runtime_default_packages") {
+				old, new := rd.GetChange("runtime_default_packages")
+				oldMap, newMap := old.(map[string]interface{}), new.(map[string]interface{})
+
+				// Packages can not be removed, only added
+				if len(oldMap) > len(newMap) {
+					rd.ForceNew("runtime_default_packages")
+				}
+
+				// Packages changes not the version can not be done in place
+				for k := range oldMap {
+					if _, ok := newMap[k]; !ok {
+						rd.ForceNew("runtime_default_packages")
+					}
+				}
+			}
+
+			return nil
+		},
+		Timeout: 5 * time.Minute,
+	}
+}
+
+func (m AutomationRuntimeEnvironmentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
```

Comment:
Schema fields are not properly ordered

Current Order:

name
resource_group_name
automation_account_name
runtime_language
runtime_version
runtime_default_packages
location
description
tags

Issue: Fields should follow the standard Azure ordering: name, resource_group_name, location as the first three required fields, then other required fields alphabetically, then optional fields alphabetically with tags last.

```suggestion
// Required fields first (standard Azure order)
"name": {...},
"resource_group_name": commonschema.ResourceGroupName(),
"location": commonschema.Location(),

// Other required fields (alphabetical)
"automation_account_name": {...},
"runtime_language": {...},
"runtime_version": {...},

// Optional fields (alphabetical, tags last)
"description": {...},
"runtime_default_packages": {...},
"tags": commonschema.Tags(),
```

Reply:
Note the comments in the suggested code need not to be included